File: filerenamer-frontend/src/main/java/nl/maas/filerenamer/frontend/WicketApplication.java
package nl.maas.filerenamer.frontend;

import nl.maas.filerenamer.frontend.wicket.pages.MainPage;
import org.apache.wicket.Page;
import org.apache.wicket.protocol.http.WebApplication;
import org.apache.wicket.resource.FileSystemResourceReference;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;

import java.nio.file.Path;

@SpringBootApplication
public class WicketApplication extends WebApplication {

    public static void main(String[] args) throws Exception {
        SpringApplication.run(WicketApplication.class, args);
    }

    public static void restart() {
        final ConfigurableApplicationContext[] ctx = {ContextProvider.ctx};
        ApplicationArguments args = ctx[0].getBean(ApplicationArguments.class);
        BrowserManager manager = ctx[0].getBean(BrowserManager.class);
        manager.close();
        Thread thread = new Thread(() -> {
            ctx[0].close();
            ctx[0] = SpringApplication.run(WicketApplication.class, args.getSourceArgs());
        });

        thread.setDaemon(false);
        thread.start();
    }

    public WicketApplication() {
    }

    @Override
    protected void internalInit() {
        super.internalInit();
        FileSystemResourceReference favicon = new FileSystemResourceReference("favicon", Path.of(this.getClass().getResource("/open/images/icon.png").getPath()));
        mountResource("/images/icon.png", favicon);
    }

    @Override
    public Class<? extends Page> getHomePage() {
        return MainPage.class;
    }
}

File: filerenamer-frontend/src/main/kotlin/nl/maas/filerenamer/frontend/WicketApplication.kt
package nl.maas.filerenamer.frontend

import nl.maas.filerenamer.frontend.wicket.pages.MainPage
import org.apache.wicket.Page
import org.apache.wicket.protocol.http.WebApplication
import org.apache.wicket.resource.FileSystemResourceReference
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import java.nio.file.Path

@SpringBootApplication
class WicketApplication : WebApplication() {
    override fun internalInit() {
        super.internalInit()
        val favicon =
            FileSystemResourceReference("favicon", Path.of(this.javaClass.getResource("/open/images/icon.png").path))
        mountResource("/images/icon.png", favicon)
        val ctx = arrayOf(ContextProvider.ctx)
    }

    override fun getHomePage(): Class<out Page?> {
        return MainPage::class.java
    }

//    companion object {
//
//        fun restart() {
//            val ctx = arrayOf(ContextProvider.ctx)
//            val args = ctx[0].getBean(ApplicationArguments::class.java)
//            val manager: BrowserManager = ctx[0].getBean(BrowserManager::class.java)
//            manager.close()
//            val thread = Thread {
//                ctx[0].close()
//                ctx[0] = SpringApplication.run(WicketApplication::class.java, *args.sourceArgs)
//            }
//
//            thread.isDaemon = false
//            thread.start()
//        }
//    }
}

fun main(args: Array<String>) {
    SpringApplication.run(WicketApplication::class.java, *args)
}
